src/components/application_manager/test/message_helper/message_helper_test.cc

```diff
@@ -2081,6 +2246,298 @@ TEST_F(MessageHelperTest, SendGetListOfPermissionsResponse_SUCCESS) {
+}
+
+TEST_F(
+    MessageHelperTest,
+    SendGetUserFriendlyMessageResponseWithMessages_ExpectSendCorrectMessageToHMI) {
+  policy::UserFriendlyMessage arr[] = {
+      {"first_code"}, {"second_code"}, {"third_code"}};
+  std::vector<policy::UserFriendlyMessage> msg(arr, arr + 3);
+  CheckSendingGetUserFriendlyMessage(msg);
+}
+
+TEST_F(MessageHelperTest,
+       SendGlobalPropertiesToHMIApplicationIsNotValid_ExpectReturnFromMethod) {
+  MockApplicationSharedPtr app;
+  EXPECT_CALL(mock_application_manager_, ManageHMICommand(_)).Times(0);
+  MessageHelper::SendGlobalPropertiesToHMI(app, mock_application_manager_);
+}
+
+TEST_F(
+    MessageHelperTest,
+    SendGlobalPropertiesToHMIWithUIGlobalProperties_ExpectSendCorrectMessageToHMI) {
+  MockApplicationSharedPtr app = ::utils::MakeShared<AppMock>();
+  const uint8_t number_msg_params = 6;
+  EXPECT_CALL(mock_application_manager_, GetNextHMICorrelationID())
+      .WillOnce(Return(kCorrelationId));
+  smart_objects::SmartObject dummy_so(smart_objects::SmartType_Map);
+  dummy_so["dummy_param"] = "dummy_param";
+  EXPECT_CALL(*app, vr_help_title()).Times(3).WillRepeatedly(Return(&dummy_so));
+  EXPECT_CALL(*app, vr_help()).Times(2).WillRepeatedly(Return(&dummy_so));
+  EXPECT_CALL(*app, keyboard_props())
+      .Times(2)
+      .WillRepeatedly(Return(&dummy_so));
+  EXPECT_CALL(*app, menu_title()).Times(2).WillRepeatedly(Return(&dummy_so));
+  EXPECT_CALL(*app, menu_icon()).Times(2).WillRepeatedly(Return(&dummy_so));
+  EXPECT_CALL(*app, app_id()).WillOnce(Return(kAppId));
+  EXPECT_CALL(*app, help_prompt())
+      .WillOnce(Return(static_cast<smart_objects::SmartObject*>(NULL)));
+  EXPECT_CALL(*app, timeout_prompt())
+      .WillOnce(Return(static_cast<smart_objects::SmartObject*>(NULL)));
+
+  smart_objects::SmartObjectSPtr result;
+  EXPECT_CALL(mock_application_manager_, ManageHMICommand(_))
+      .WillOnce(DoAll(SaveArg<0>(&result), Return(true)));
+  MessageHelper::SendGlobalPropertiesToHMI(app, mock_application_manager_);
+
+  ASSERT_TRUE(result.valid());
+  smart_objects::SmartObject& global_properties_msg_params =
+      (*result)[strings::msg_params];
+
+  EXPECT_EQ((*result)[strings::params][strings::function_id].asInt(),
+            hmi_apis::FunctionID::UI_SetGlobalProperties);
+  CheckParamsOfMessage(result, true, true);
+
+  EXPECT_EQ(global_properties_msg_params.length(), number_msg_params);
+  EXPECT_TRUE(global_properties_msg_params.keyExists(strings::vr_help_title));
+  EXPECT_TRUE(global_properties_msg_params.keyExists(strings::vr_help));
+  EXPECT_TRUE(
+      global_properties_msg_params.keyExists(strings::keyboard_properties));
+  EXPECT_TRUE(global_properties_msg_params.keyExists(strings::menu_title));
+  EXPECT_TRUE(global_properties_msg_params.keyExists(strings::menu_icon));
+  std::set<std::string> keys = global_properties_msg_params.enumerate();
+  keys.erase(strings::app_id);
+  for (std::set<std::string>::iterator it = keys.begin(); it != keys.end();
```

Comment:
@dev-gh Here I check value in SO by key. And I don't understand how find_if can help me.